JMS/com/ibm/mq/samples/jms/JmsResponse.java

```diff
@@ -131,6 +132,7 @@ private static void replyToMessage(JMSContext context, Message receivedMessage,
                 }
             }
 
+            context.rollback();

```

Comment:
rollback should always happen in the catch block, the various returns make this optional which doesn't seem right. Also MQRC_UNKNOWN_OBJECT_NAME is both handled as a Exception and a RTE. That doesn't seem right.